src/NLog/LayoutRenderers/NdcLayoutRenderer.cs

```diff
@@ -77,7 +77,7 @@ public NdcLayoutRenderer()
         /// <param name="logEvent">Logging event.</param>
         protected override void Append(StringBuilder builder, LogEventInfo logEvent)
         {
-            string[] messages = NestedDiagnosticsContext.GetAllMessages();
+            object[] messages = NestedDiagnosticsContext.GetAllObjects();
```

Comment:
Added a GetAllMessages() overload that accepts an `IFormatProvider` and used it.